append_lifecycle.go

```diff
@@ -407,20 +406,23 @@ func (i *integrationStats) statsDecorator(delegate AddFn) AddFn {
 		return func() (Index, error) {
 			idx, err := f()
 			attr := []attribute.KeyValue{}
-			pushbackType := "" // This will be used for the pushback attribute below, empty string means no pushback
 
 			if err != nil {
-				if errors.Is(err, ErrPushback) {
-					// record the the fact there was pushback, and use the error string as the type.
-					pushbackType = err.Error()
-				} else {
-					// Just flag that it's an errored request to avoid high cardinality of attribute values.
+				switch {
+				// Record the fact there was pushback, if any.
+				case errors.Is(err, ErrPushbackAntispam):
+					attr = append(attr, attribute.String("tessera.pushback", "antispam"))
+				case errors.Is(err, ErrPushbackIntegration):
+					attr = append(attr, attribute.String("tessera.pushback", "integration"))
+				case errors.Is(err, ErrPushback):
+					attr = append(attr, attribute.String("tessera.pushback", "other"))
+				// If it's not a pushback, just flag that it's an errored request to avoid high cardinality of attribute values.
```

Comment:
I think this should be inside the `default:`?

Reply:
I moved it back. I had put it there in the first place to match the indentation of the other comment, that itself applied to 3 cases.